commit d3b1a2d5b3515b53d210de0a0535acaf3cf4c2a7
Author: Lionel Elie Mamane <lionel@mamane.lu>
Date:   Mon Aug 15 11:44:11 2011 +0200

    [PATCH] fdo#35784 minimal fix
    
    Fake DatabaseMetaData::getImportedKeys: we can get away with not giving column names if we give zero data rows

diff --git a/patches/postgresql/sdbc-postgresql.diff b/patches/postgresql/sdbc-postgresql.diff
index 0b978e3..649d1c0 100644
--- libreoffice-3.3.3/libreoffice-build/patches/postgresql/sdbc-postgresql.diff
+++ libreoffice-3.3.3/libreoffice-build/patches/postgresql/sdbc-postgresql.diff
@@ -2955,7 +2955,7 @@ new file mode 100644
 index 0000000..b3c18d8
 --- /dev/null
 +++ connectivity/source/drivers/postgresql/pq_databasemetadata.cxx
-@@ -0,0 +1,2164 @@
+@@ -0,0 +1,2162 @@
 +/*************************************************************************
 + *
 + *  $RCSfile: pq_databasemetadata.cxx,v $
@@ -4654,11 +4654,9 @@ index 0000000..b3c18d8
 +//             makeAny( resolveColumnNames(foreignMap, xRow->getString(8) ) ) );
 +
 +//     }
-+    // fake the getImportedKey() function call in
-+    // dbaccess/source/ui/relationdesigin/RelationController.cxx
-+    // it seems to be the only place in the office, where this function is needed
++    // LEM TODO: full "real" implementation
 +    return new SequenceResultSet(
-+        m_refMutex, *this, Sequence< OUString >(), Sequence< Sequence< Any > > (1), m_pSettings->tc );
++        m_refMutex, *this, Sequence< OUString >(), Sequence< Sequence< Any > > (), m_pSettings->tc );
 +}
 +
 +::com::sun::star::uno::Reference< XResultSet > DatabaseMetaData::getExportedKeys(
